### PR TITLE
add insights_id to /systems endpoint

### PR DESCRIPTION
**Pull request by mkholjuraev**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,10 +8,13 @@ require (
 	github.com/RedHatInsights/patchman-clients/vmaas v0.14.0
 	github.com/aws/aws-sdk-go v1.30.9
 	github.com/cenkalti/backoff v2.2.1+incompatible // indirect
+	github.com/cpuguy83/go-md2man/v2 v2.0.0 // indirect
```
Comment (Josca) [inline go.mod:11]: Do we need some deps update for that task? I don't think so.

File: vmaas_sync/metrics_db_test.go [hunk 2]
```diff
@@ -41,5 +41,5 @@ func TestDatabaseProcCounts(t *testing.T) {
 	processesInfo := getDatabaseProcesses()
 
 	assert.Less(t, 0, len(processesInfo))
-	assert.Equal(t, "-", processesInfo[0].Key)
+	assert.Equal(t, "admin", processesInfo[0].Key)
```
Comment (Josca) [inline vmaas_sync/metrics_db_test.go:44]: That will be the reason TestDatabaseProcCounts fails.

File: vmaas_sync/system_culling_test.go [hunk 3]
```diff
@@ -143,11 +143,13 @@ func TestCullSystems(t *testing.T) {
 	nToDelete := 4
 	for i := 0; i < nToDelete; i++ {
 		invID := fmt.Sprintf("00000000-0000-0000-0000-000000000de%d", i+1)
+		insightsID := fmt.Sprintf("00000000-0000-0000-0000-000000000de%d", i+1)
```
Comment (Josca) [inline vmaas_sync/system_culling_test.go:146]: Isn't that `insightsID = invID`?

**Review by Josca — CHANGES_REQUESTED** [review]
Ok, thanks, some things to update suggested.

File: vmaas_sync/system_culling_test.go (hunk 3 above)
Comment (MichaelMraka) [reply vmaas_sync/system_culling_test.go:146]: What's insights_id? Isn't it actually inventory_id which is already part of system_platform table?

Comment (mkholjuraev) [reply vmaas_sync/system_culling_test.go:146]: @MichaelMraka maybe it is the case. insights_id was provided by the inventory UI app which is needed for displaying the Inventory table. My assumption was that insights_id is synced when a new system is uploaded/registered. Thus, I have tried to sync this info also together with other system information into the system_platform.  Please have a look at the picture and let me know if it is already in the system_platform as inventory_id

x-special/nautilus-clipboard
copy
file:///home/muslimjon/Pictures/insights_id.png

Comment (MichaelMraka) [reply vmaas_sync/system_culling_test.go:146]: I see, it's yet another different id :( . Inventory_id is referred as 'id' on your picture.

Comment (MichaelMraka) [reply vmaas_sync/system_culling_test.go:146]: So you have it right you need to store it as a new column. Just please for the test use different value than system's inventory_id, e.g "00000000-0000-0000-1111-000000000de%d".

File: dev/test_data.sql [hunk 4]
```diff
@@ -15,27 +15,27 @@ DELETE FROM strings;
...
-(13, '00000000-0000-0000-0000-000000000013','00000000-0000-0000-0000-000000000013', 3, '{ "package_list": [ "kernel-2.6.32-696.20.1.el6.x86_64" ]}', '1', '2018-09-22 12:00:00-04', '2018-01-22 12:00:00-04', 1,0),
-(14, '00000000-0000-0000-0000-000000000014','00000000-0000-0000-0000-000000000014', 3, '{ "package_list": [ "kernel-2.6.32-696.20.1.el6.x86_64" ]}', '1', '2018-09-22 12:00:00-04', '2018-01-22 12:00:00-04', 0,0);
+INSERT INTO system_platform (id, inventory_id, display_name, rh_account_id, reporter_id, vmaas_json, json_checksum, last_evaluation, last_upload, packages_installed, packages_updatable, third_party,insights_id) VALUES
+(1, '00000000-0000-0000-0000-000000000001','00000000-0000-0000-0000-000000000001', 1, 1, '{ "package_list": [ "kernel-2.6.32-696.20.1.el6.x86_64" ]}', '1', '2018-09-22 12:00:00-04', '2020-09-22 12:00:00-04',0,0, true, '00000000-0000-0000-0000-000000000001'),
+(2, '00000000-0000-0000-0000-000000000002','00000000-0000-0000-0000-000000000002', 1, 1, '{ "package_list": [ "kernel-2.6.32-696.20.1.el6.x86_64" ]}', '1', '2018-09-22 12:00:00-04', '2018-09-22 12:00:00-04',0,0, false, '00000000-0000-0000-0000-000000000002'),
```
Comment (MichaelMraka) [inline dev/test_data.sql:20]: Please use different inventory_id here, e.g. '00000000-0000-0000-1111-000000000002' could be a good choice :).

File: listener/common_test.go [hunk 5]
```diff
@@ -13,6 +13,7 @@ import (
 )
 
 const id = "99c0ffee-0000-0000-0000-0000c0ffee99"
+const insightsID = "00000000-0000-0000-0000-000000000001"
```
Comment (MichaelMraka) [inline listener/common_test.go:16]: And here as well ("99c0ffee-0000-0000-1111-0000c0ffee99").

File: manager/controllers/system_delete_test.go [hunk 6]
```diff
@@ -12,6 +12,7 @@ import (
 )
 
 const del = "99c0ffee-0000-0000-0000-000000000de1"
+const insightsID = "00000000-0000-0000-0000-000000000001"
```
Comment (MichaelMraka) [inline manager/controllers/system_delete_test.go:15]: And here  ("99c0ffee-0000-0000-1111-000000000de1").

File: manager/controllers/systems_export_test.go [hunk 7]
```diff
@@ -56,12 +56,12 @@ func TestSystemsExportCSV(t *testing.T) {
...
 	assert.Equal(t,
 		"id,display_name,last_evaluation,last_upload,rhsa_count,rhba_count,rhea_count,stale,"+
-			"third_party,packages_installed,packages_updatable,os_name,os_major,os_minor,rhsm,"+
+			"third_party,insights_id,packages_installed,packages_updatable,os_name,os_major,os_minor,rhsm,"+
 			"stale_timestamp,stale_warning_timestamp,culled_timestamp,created",
 		lines[0])
 
 	assert.Equal(t, "00000000-0000-0000-0000-000000000001,00000000-0000-0000-0000-000000000001,"+
-		"2018-09-22T16:00:00Z,2020-09-22T16:00:00Z,2,3,3,false,true,0,0,RHEL,8,1,8.1,"+
+		"2018-09-22T16:00:00Z,2020-09-22T16:00:00Z,2,3,3,false,true,00000000-0000-0000-0000-000000000001,0,0,RHEL,8,1,8.1,"+
```
Comment (MichaelMraka) [inline manager/controllers/systems_export_test.go:64]: And here (0000000-0000-0000-1111-000000000001).

**Review by MichaelMraka — CHANGES_REQUESTED** [review]
I've seen a lot of code where people interchanged one kind of id for the other and tests passed but real app (unsurprisingly) the failed :).

**Review by MichaelMraka — APPROVED** [review]

**Comment by mkholjuraev** [comment]
I am curious what will happen with already existing systems in the system_platform? Most probably they will not have insights_id added after this PR, right? Because this PR only manages newly uploaded systems.

**Pull request closed without merging**